lab_2_tokenize_by_bpe/main.py

```diff
@@ -81,7 +156,24 @@ def decode(
     :param end_of_word_token: an end-of-word token
     :return: decoded sequence
     """
-
+    if not isinstance(encoded_text, list) or\
+            not isinstance(vocabulary, dict) or\
+            not (isinstance(end_of_word_token, str) or end_of_word_token is None):
+        return None
+    decoding = ''
+    for number in encoded_text:
+        for token in vocabulary:
```

Comment:
Consider simply inverting the dictionary

Reply:
Done